src/mapbuffer.cpp

```diff
@@ -55,6 +55,19 @@ void mapbuffer::clear()
     submaps.clear();
 }
 
+void mapbuffer::clear_outside_reality_bubble()
+{
+    map &here = get_map();
+    auto it = submaps.begin();
+    while( it != submaps.end() ) {
```

Comment:
Kinda mad the `std::map` specialization of `erase_if` is c++20.

Reply:
We could easily write our own implementation an throw it in `cata_utility.h`.  But not in this PR, I think.